src/arbcalls/arb_poly.jl

```diff
@@ -0,0 +1,287 @@
+arbcall"void arb_poly_revert_series_lagrange(arb_poly_t h, const arb_poly_t f, slong n, slong prec)"
+arbcall"void _arb_poly_revert_series_newton(arb_ptr h, arb_srcptr f, slong flen, slong n, slong prec)"
+arbcall"void arb_poly_revert_series_newton(arb_poly_t h, const arb_poly_t f, slong n, slong prec)"
+arbcall"void _arb_poly_revert_series_lagrange_fast(arb_ptr h, arb_srcptr f, slong flen, slong n, slong prec)"
+arbcall"void arb_poly_revert_series_lagrange_fast(arb_poly_t h, const arb_poly_t f, slong n, slong prec)"
+arbcall"void _arb_poly_revert_series(arb_ptr h, arb_srcptr f, slong flen, slong n, slong prec)"
+arbcall"void arb_poly_revert_series(arb_poly_t h, const arb_poly_t f, slong n, slong prec)"
+
+### Evaluation
+arbcall"void _arb_poly_evaluate_horner(arb_t y, arb_srcptr f, slong len, const arb_t x, slong prec)"
+arbcall"void arb_poly_evaluate_horner(arb_t y, const arb_poly_t f, const arb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate_rectangular(arb_t y, arb_srcptr f, slong len, const arb_t x, slong prec)"
+arbcall"void arb_poly_evaluate_rectangular(arb_t y, const arb_poly_t f, const arb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate(arb_t y, arb_srcptr f, slong len, const arb_t x, slong prec)"
+arbcall"void arb_poly_evaluate(arb_t y, const arb_poly_t f, const arb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate_acb_horner(acb_t y, arb_srcptr f, slong len, const acb_t x, slong prec)"
+arbcall"void arb_poly_evaluate_acb_horner(acb_t y, const arb_poly_t f, const acb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate_acb_rectangular(acb_t y, arb_srcptr f, slong len, const acb_t x, slong prec)"
+arbcall"void arb_poly_evaluate_acb_rectangular(acb_t y, const arb_poly_t f, const acb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate_acb(acb_t y, arb_srcptr f, slong len, const acb_t x, slong prec)"
+arbcall"void arb_poly_evaluate_acb(acb_t y, const arb_poly_t f, const acb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate2_horner(arb_t y, arb_t z, arb_srcptr f, slong len, const arb_t x, slong prec)"
+arbcall"void arb_poly_evaluate2_horner(arb_t y, arb_t z, const arb_poly_t f, const arb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate2_rectangular(arb_t y, arb_t z, arb_srcptr f, slong len, const arb_t x, slong prec)"
+arbcall"void arb_poly_evaluate2_rectangular(arb_t y, arb_t z, const arb_poly_t f, const arb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate2(arb_t y, arb_t z, arb_srcptr f, slong len, const arb_t x, slong prec)"
+arbcall"void arb_poly_evaluate2(arb_t y, arb_t z, const arb_poly_t f, const arb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate2_acb_horner(acb_t y, acb_t z, arb_srcptr f, slong len, const acb_t x, slong prec)"
+arbcall"void arb_poly_evaluate2_acb_horner(acb_t y, acb_t z, const arb_poly_t f, const acb_t x, slong prec)"
+arbcall"void _arb_poly_evaluate2_acb_rectangular(acb_t y, acb_t z, arb_srcptr f, slong len, const acb_t x, slong prec)"
```

Comment:
Here we should probably change the code gen to produce a function which also accepts an `ArbVector` for the two arguments `arb_srcptr f, slong len`

Reply:
My plan here was to later implement methods which are not in-place for which we add more conveniences, like automatically picking up the length from `AcbVector` and such. It would be great if we could change the code-gen to automatically make length a keyword argument which takes the length of `AcbVector` as default value, but I'm worried this would be hard to do. The length parameter is sometimes called `len`, sometimes `flen` or `glen`, so it's hard to automatically identify it.

Reply:
Makes sense, but note that I handled some code exactly doing this

> It would be great if we could change the code-gen to automatically make length a keyword argument which takes the length of AcbVector as default value

in #27 for the vector types 🙃